clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesNamedAccountCredentials.java

```diff
@@ -167,6 +171,20 @@ public Permissions getPermissions() {
     return permissions;
   }
 
+  public Map<String, String> getSpinnakerKindMap() {
+    if (kubernetesSpinnakerKindMap == null) {
+      return new HashMap<String, String>();
+    }
+    Map<String, String> kindMap = kubernetesSpinnakerKindMap.kubernetesToSpinnakerKindStringMap();
```

Comment:
Maybe we should make a copy of this? I think by modifying it below we modify the global copy

Reply:
```java
Map<String, String> kindMap = new HashMap<>(kubernetesSpinnakerKindMap.kubernetesToSpinnakerKindStringMap());